app/controllers/feeds_controller.rb

```diff
@@ -46,10 +53,15 @@ def update
 
   # DELETE /feeds/1
   def destroy
-    @feed.destroy
-
     respond_to do |format|
-      format.html { redirect_to feeds_url, notice: "Feed was successfully destroyed." }
+      if @feed.destroy
+        format.html { redirect_to podcast_feed_path(@podcast, @podcast.default_feed), notice: (t ".success", model: "Feed") }
+      else
+        format.html do
+          flash.alert = t ".failure", model: "Feed"
```

Comment:
I think these alerts in here need to be the `.now` flash style - otherwise you get 2 alerts.  Once after you click save, and a second one if you click away.

Generally, if you're doing a `render`, you need to use `flash.now`.  If you're redirecting, you use the regular `flash` (or append the `notice:` to the `redirect_to` itself... which amounts to the same thing).